### PR TITLE
Loop and test restructuring

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Makes the following changes

```diff
@@ -4,9 +4,7 @@
 │   │   ├── base.py
 │   │   ├── batch
 │   │   │   ├── __init__.py
-│   │   │   ├── manual.py
 │   │   │   └── training_batch_loop.py
-│   │   ├── closure.py
 │   │   ├── dataloader
 │   │   │   ├── __init__.py
 │   │   │   ├── dataloader_loop.py
@@ -18,8 +16,10 @@
 │   │   │   ├── prediction_epoch_loop.py
 │   │   │   └── training_epoch_loop.py
 │   │   ├── fit_loop.py
-│   │   ├── optimizer
+│   │   ├── optimization
 │   │   │   ├── __init__.py
+│   │   │   ├── closure.py
+│   │   │   ├── manual_loop.py
 │   │   │   └── optimizer_loop.py
 │   │   └── utilities.py
 └── tests
@@ -28,16 +28,15 @@
     │   ├── batch
     │   │   ├── __init__.py
     │   │   └── test_truncated_bptt.py
-    │   ├── test_closure.py
-    │   ├── test_loop_state_dict.py
-    │   └── test_loops.py
-    ├── trainer
-    │   ├── loops
+    │   ├── optimization
     │   │   ├── __init__.py
-    │   │   ├── test_all.py
-    │   │   ├── test_evaluation_loop.py
-    │   │   ├── test_evaluation_loop_flow.py
-    │   │   ├── test_flow_warnings.py
-    │   │   ├── test_training_loop.py
-    │   │   ├── test_training_loop_flow_dict.py
-    │   │   └── test_training_loop_flow_scalar.py
+    │   │   └── test_closure.py
+    │   ├── test_all.py
+    │   ├── test_evaluation_loop.py
+    │   ├── test_evaluation_loop_flow.py
+    │   ├── test_flow_warnings.py
+    │   ├── test_loop_state_dict.py
+    │   ├── test_loops.py
+    │   ├── test_training_loop.py
+    │   ├── test_training_loop_flow_dict.py
+    │   └── test_training_loop_flow_scalar.py
```

### Does your PR introduce any breaking changes? If yes, please list them.

None user facing

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]
